src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -4315,20 +4318,38 @@ class C
 ref struct S
 {
     public ref long M(ref long x) => ref x;
-}").VerifyDiagnostics(
+}";
+
+            var comp = CreateCompilationWithMscorlibAndSpan(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(
                 // (8,20): error CS8157: Cannot return 'y' by reference because it was initialized to a value that cannot be returned by reference
                 //         return ref y;
                 Diagnostic(ErrorCode.ERR_RefReturnNonreturnableLocal, "y").WithArguments("y").WithLocation(8, 20),
                 // (16,24): error CS8157: Cannot return 'y' by reference because it was initialized to a value that cannot be returned by reference
                 //             return ref y;
                 Diagnostic(ErrorCode.ERR_RefReturnNonreturnableLocal, "y").WithArguments("y").WithLocation(16, 24));
+
+            comp = CreateCompilationWithMscorlibAndSpan(source);
+            comp.VerifyDiagnostics(
+                // (7,26): error CS8350: This combination of arguments to 'S.M(ref long)' is disallowed because it may expose variables referenced by parameter 'x' outside of their declaration scope
```

Comment:
I think this error would be easier to understand if the parameter `ref long x` had a different name than the local `long x`.

Reply:
I don't think I understand why the new diagnostics are correct compared to the old diagnostics. The only rule change here should be that the `this` parameter in `S.M` is now `scoped ref` when it was just `ref` before. We're saying a ref to `receiver` will not escape now--why does that make this call problematic, then?